app/controllers/users_controller.rb

```diff
@@ -1,9 +1,40 @@
 class UsersController < ApplicationController
-  before_action :load_user, only: %i(show)
+  before_action :logged_in_user, only: %i(index edit update destroy)
+  before_action :load_user, only: %i(show edit update destroy)
+  before_action :correct_user, only: %i(edit update)
+  before_action :admin_user, only: :destroy
 
   # GET /users/:id
   def show; end
 
+  # GET /users/:id/edit
+  def edit; end
+
+  # PATCH /users/:id
+  def update
+    if @user.update(user_params)
+      flash[:success] = t("flash.update_success")
+      redirect_to @user
+    else
+      render :edit, status: :unprocessable_entity
+    end
+  end
+
+  # GET /users
+  def index
+    @pagy, @users = pagy User.all, items: Settings.default_page_items
```

Comment:
```suggestion
    @pagy, @users = pagy User.all, items: Settings.default_page_items
```
Thêm scope để sort users theo thứ tự mới nhất nữa nha